### ISSUE TITLE
Dpcpp compilation warning: Unknown CUDA version 10.2

### ISSUE THREAD

**Issue by mmmm1998** [intel/llvm#2271]
I am trying DPC++ with CUDA and I have built dpcpp with CUDA support (HEAD is 0b8947a9dbffb6911a5cd00b2cad98d885747c78). But on compilation I see this warnings
````
$ clang++ -fsycl -fsycl-targets=nvptx64-nvidia-cuda-sycldevice main.cpp vector-add.cpp -o vector-add
clang-12: warning: Unknown CUDA version 10.2. Assuming the latest supported version 10.1 [-Wunknown-cuda-version]
````
And compiled binaries don't see GPU (can't found available devices for gpu_selector). Do I do something wrong?

This is my system info:
````
$ nvidia-smi
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.33.01    Driver Version: 440.33.01    CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 1080    On   | 00000000:01:00.0 Off |                  N/A |
| 28%   39C    P8    12W / 180W |      1MiB /  8116MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
````
````
$ lsb_release -a
LSB Version:    core-9.20170808ubuntu1-noarch:printing-9.20170808ubuntu1-noarch:security-9.20170808ubuntu1-noarch
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.4 LTS
Release:        18.04
Codename:       bionic
````

**Comment by vladimirlaz** [comment]
Hi, @mmmm1998 
The warning on compilation time does not impacts device availability.

You can check that proper SYCL backend is loaded (will GPU device be found if you set SYCL_BE=PI_CUDA to select CUDA BE). You can use SYCL_PI_TRACE environment variable to detect the problem (e.g. SYCL_PI_TRACE=1). See description for the variables in https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md.

**Comment by krasznaa** [comment]
Hi @vladimirlaz,

Could that compilation warning be silenced though? I've been sticking to using CUDA 10.1 with my DPC\+\+ installations for exactly this reason. (The build does work fine with CUDA 10.2, but I don't want our builds to print meaningless warnings.)

Of course support for CUDA 11 would also be nice. :wink: But I realise that that's probably a non-negligible amount of work. While silencing a warning about an already supported CUDA version should hopefully be a lot easier.

Cheers,
     Attila

**Comment by bader** [comment]
I think -Wno-unknown-cuda-version will silence this warning.

**Comment by npmiller** [comment]
As far as the warning goes, this has recently been updated upstream to CUDA 11.4:

* https://github.com/intel/llvm/commit/ce4545db1d31f447bb42987099d691d5658da4bf

So you should no longer see it on 10.2, and since 11.4 is currently the latest version it shouldn't be an issue at all.

However even with this warning DPC++ has no issue working with NVidia GPUs, so I'm not sure what was the original issue.

**Comment by npmiller** [comment]
It has been a while, I'm not able to reproduce the original issue, and the warning should no longer be there, so I'm closing this ticket.

@mmmm1998 feel free to re-open if this is still a problem.
